### PR TITLE
Update and improve gVisor support:

### PR DESCRIPTION
**Pull request by jcpittman144**
- Implement event handlers for recently-added gVisor native events, including WRITE
- Enhance event handlers to explicitly document which SCAP event fields are invalid (e.g. where info is not provided by the native gVisor event)
- Improve gvisor config generator, to configure all fields needed by the event parser, and only those fields
- Zero fill allocated scap_gvisor_platform structure, to avoid crash on startup
- Return SCAP_TIMEOUT instead of SCAP_EOF on interrupted socket read
- Add new gvisor parser unit tests for new event handlers; and fix existing unit tests to reflect updated functionality of existing event handlers

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

/area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

/area libsinsp

/area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:
Enhance and update gVisor support

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix: Enhance and update gVisor support, including additional system call events
```

### CODE REVIEW

File: userspace/libscap/engine/gvisor/parsers.cpp
Comment (dwindsor) [inline userspace/libscap/engine/gvisor/parsers.cpp:182]: Nit: this is actually different from the `cwd` value (`""`)  in the previous call to `scap_event_encode_params`, which also IIUC encodes process data for `init`.

**Comment by LucaGuerra** [comment]
This is massive! Thanks for adding tests, these are really important if we ever want to refactor any of it since it's a lot of code :pray: .
Re. the "filler" structure we can see if it's the right thing for us and change it later. We definitely want these improvements in the next version of libs. Please take a look at @dwindsor 's comment above but I can already approve this. Thanks!

**Review by LucaGuerra — DISMISSED** [review]

**Comment by LucaGuerra** [comment]
/hold

@jcpittman144 once you address review comments please simply add a message with slash unhold to merge the PR

File: userspace/libscap/engine/gvisor/parsers.cpp
Comment (jcpittman144) [reply userspace/libscap/engine/gvisor/parsers.cpp:182]: Good catch.
I thought I was doing the right thing, by hardcoding the current working directory as "/", for the container start event.  But now I'm not so sure.
I will restore the original value (empty string).

**Comment by jcpittman144** [comment]
@LucaGuerra, @dwindsor,
I addressed the review comment, and made a change to restore the original behavior of one of the generated events.
That change invalidated the LGTM designation.
So the PR is blocked again, waiting for an approval.
If everything looks good, please approve and I will do the merge.
Thanks!

**Review by LucaGuerra — APPROVED** [review]
Thank you!

**Review by leogr — APPROVED** [review]

**Pull request merged**
